Realm/Tests/ObjectTests.m

```diff
@@ -1002,6 +1002,27 @@ - (void)testIndex
     XCTAssertFalse(ageProperty.indexed, @"non-indexed property shouldn't have an index");
 }
 
+- (void)testIndexedPropertyRejectsNullBytesInString
+{
+    RLMRealm *realm = RLMRealm.defaultRealm;
+    [realm beginWriteTransaction];
+    IndexedObject *io = [IndexedObject createInDefaultRealmWithObject:@[@"", @0]];
+    XCTAssertThrows(io.name = @"a\0b");
+    XCTAssertThrows(([IndexedObject createInDefaultRealmWithObject:@[@"a\0b", @0]]));
```

Comment:
we should probably also test that an exception is thrown when adding a standalone object with an invalid string property